pkg/utils/resources/suite_test.go

```diff
@@ -0,0 +1,409 @@
+				},
+
+				InitContainers: []v1.Container{
+					{
+						RestartPolicy: lo.ToPtr(v1.ContainerRestartPolicyAlways),
+						Resources: v1.ResourceRequirements{
+							Limits:   v1.ResourceList{v1.ResourceCPU: resource.MustParse("4"), v1.ResourceMemory: resource.MustParse("2Gi")},
+							Requests: v1.ResourceList{v1.ResourceCPU: resource.MustParse("1"), v1.ResourceMemory: resource.MustParse("2Gi")},
+						},
+					},
+					{
+						Resources: v1.ResourceRequirements{
+							Limits:   v1.ResourceList{v1.ResourceCPU: resource.MustParse("10"), v1.ResourceMemory: resource.MustParse("2Gi")},
+							Requests: v1.ResourceList{v1.ResourceCPU: resource.MustParse("1"), v1.ResourceMemory: resource.MustParse("2Gi")},
+						},
+					},
+				},
+			})
+			podResources := Ceiling(pod)
+			Expect(podResources.Requests.Cpu().Value()).To(BeNumerically("==", 3))
+			Expect(podResources.Requests.Memory().Value()).To(BeNumerically("==", 4294967296))
+			Expect(podResources.Limits.Cpu().Value()).To(BeNumerically("==", 14))
+			Expect(podResources.Limits.Memory().Value()).To(BeNumerically("==", 4294967296))
+		})
+
+		It("should schedule based on the max resource requests of containers and initContainers with sidecarContainers when there is a large sidecar container", func() {
+			pod := test.Pod(test.PodOptions{
+				ResourceRequirements: v1.ResourceRequirements{
+					Limits:   v1.ResourceList{v1.ResourceCPU: resource.MustParse("2"), v1.ResourceMemory: resource.MustParse("1Gi")},
+					Requests: v1.ResourceList{v1.ResourceCPU: resource.MustParse("2"), v1.ResourceMemory: resource.MustParse("1Gi")},
+				},
+
+				InitContainers: []v1.Container{
+					{
+						RestartPolicy: lo.ToPtr(v1.ContainerRestartPolicyAlways),
+						Resources: v1.ResourceRequirements{
+							Limits:   v1.ResourceList{v1.ResourceCPU: resource.MustParse("4"), v1.ResourceMemory: resource.MustParse("2Gi")},
+							Requests: v1.ResourceList{v1.ResourceCPU: resource.MustParse("1"), v1.ResourceMemory: resource.MustParse("2Gi")},
+						},
+					},
+					{
+						Resources: v1.ResourceRequirements{
+							Limits:   v1.ResourceList{v1.ResourceCPU: resource.MustParse("1"), v1.ResourceMemory: resource.MustParse("2Gi")},
+							Requests: v1.ResourceList{v1.ResourceCPU: resource.MustParse("1"), v1.ResourceMemory: resource.MustParse("2Gi")},
+						},
+					},
+				},
+			})
+			podResources := Ceiling(pod)
+			Expect(podResources.Requests.Cpu().Value()).To(BeNumerically("==", 3))
+			Expect(podResources.Requests.Memory().Value()).To(BeNumerically("==", 4294967296))
+			Expect(podResources.Limits.Cpu().Value()).To(BeNumerically("==", 6))
+			Expect(podResources.Limits.Memory().Value()).To(BeNumerically("==", 4294967296))
+		})
+
+		It("should schedule based on the max resource requests of containers and initContainers with sidecarContainers when considering multiple sidecars and a large initcontainer", func() {
```

Comment:
```suggestion
		It("should schedule based on the max resource requests of containers and initContainers with sidecarContainers when an initContainer comes after multiple sidecarContainers", func() {
```

Reply:
Then you can create another test case: "should calculate resource requests when an initContainer comes after multiple sidecarContainers that doesn't exceed container resource requests"